src/runtime/contrib/tensorrt/tensorrt_runtime.cc

```diff
@@ -211,6 +218,16 @@ class TensorRTRuntime : public JSONRuntimeBase {
       builder.AddOutput(outputs_[i], EntryID(outputs_[i]));
     }
 
+    // Allocate Device Buffers
+    if (trt_engine_cache_.count(std::make_pair(symbol_name_, batch_size_))) {
+      TensorRTEngineAndContext& engine_and_context =
+          trt_engine_cache_.at(std::make_pair(symbol_name_, batch_size_));
+      if (engine_and_context.device_buffers.size() == 0) {
+        builder.CreateDeviceBuffers(&engine_and_context);
+        return;
```

Comment:
We are building the TRT network in the TensorRTBuilder, but exiting before `BuildEngine` is called. This means the resources used by `builder` won't ever be freed (`TensorRTBuilder::CleanUp()`) needs to be called.

Reply:
We also shouldnt have to rebuild the whole nextwork just to allocate the buffers.

Reply:
`CleanUp` is added.

In the `BuildEngine` function from `tensorrt_runtime.cc`, the whole network has not been actually rebuilt since the function returns before`builder.BuildEngine` gets called.

Reply:
I think the best solution is to move `CreateDeviceBuffers` out of TensorRTBuilder and into the runtime module. That way we can call it without the unnecessary allocations and creations done by TensorRTBuilder